### DISCUSSION TITLE
DatePicker: allow edit (without pick date from combo)

### DISCUSSION THREAD

**Discussion by ctrl-alt-d** [AvaloniaUI/Avalonia#7415] [Q&A] [answered]
Hi, is there a way to allow the user to type freely the date with keyboard into DatePicker instead to pick it from conbo?

As you can see in the video, I only can change date from calendar dropdown:

https://user-images.githubusercontent.com/3105983/150686008-1dd4de33-4f77-46a7-ac71-63d232ade368.mov

**Comment by timunie** [comment, marked as answer]
Hi,

I think this should normally work if the TextBox is not readonly. You can test it in the ControlCatalog. 

Moreover you seem to have a custom Style, there may be an issue with this. 

Happy coding

Tim

**Reply by timunie** [reply]
Here is the original style, and the TextBox seems to be editable by default https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml#L104-L114

**Reply by ctrl-alt-d** [reply]
>Moreover you seem to have a custom Style, there may be an issue with this.

Yes, I'm using[ Material Avalonia](https://github.com/AvaloniaCommunity/material.avalonia) Maybe I should move the issue to them.

**Reply by timunie** [reply]
Looks like you should do that. They marked it as todo if I saw it right 

https://github.com/AvaloniaCommunity/material.avalonia/blob/dev/Material.Styles/DatePicker.xaml#L34

**Reply by ctrl-alt-d** [reply]
Awesome. Thanks @timunie !

**Comment by agnauck** [comment]
I am currently making some tests for a LOB application that requires lots of inputs.
I looks like the current version of the CalendarDatePicker allows all characters as text input.
The default of many UI frameworks for a date input on desktop looks like in the video below.

* There are 3 input groups for day, month any year. 
* usually when the control get the focus the first group gets selected and you can easily change the date portion by typing the numbers yourself or using up/down arrow keys.
* you can also use cursor keys to switch to the next input group

Is something like that currently possible with Avalonia out of the box? Or do I need to extend the CalendarDatePicker to add this functionality?

https://github.com/user-attachments/assets/281f8fc4-0617-4e3f-8ce3-68a62a81dbf2
